### PR TITLE
shell: make sure proper .env is loaded in script runner

### PR DESCRIPTION
**Pull request by nektro**
Closes https://github.com/oven-sh/bun/issues/9445

### CODE REVIEW

File: test/js/bun/shell/bunshell.test.ts [hunk 1]
```diff
@@ -831,3 +832,61 @@ function sentinelByte(buf: Uint8Array): number {
...
+    join(package_dir, "index.ts"),
+    `console.log(process.env.NODE_ENV); console.log(process.env.AWESOME);`,
+  );
+  await writeFile(join(package_dir, ".env.production"), `AWESOME=production`);
+  await writeFile(join(package_dir, ".env.development"), `AWESOME=development`);
+
+  const { stdout, stderr, exited } = Bun.spawn({
+    cmd: [bunExe(), "run", "prod"],
+    cwd: package_dir,
+    stdio: ["pipe", "pipe", "pipe"],
```
Comment (Jarred-Sumner) [inline test/js/bun/shell/bunshell.test.ts:852]:
```suggestion
    stdio: ["ignore", "pipe", "pipe"],
```

File: test/js/bun/shell/bunshell.test.ts [hunk 2]
```diff
@@ -831,3 +832,61 @@ function sentinelByte(buf: Uint8Array): number {
...
+    join(package_dir, "index.ts"),
+    `console.log(process.env.NODE_ENV); console.log(process.env.AWESOME);`,
+  );
+  await writeFile(join(package_dir, ".env.production"), `AWESOME=production`);
+  await writeFile(join(package_dir, ".env.development"), `AWESOME=development`);
+
+  const { stdout, stderr, exited } = Bun.spawn({
+    cmd: [bunExe(), "prod"],
+    cwd: package_dir,
+    stdio: ["pipe", "pipe", "pipe"],
```
Comment (Jarred-Sumner) [inline test/js/bun/shell/bunshell.test.ts:881]:
```suggestion
    stdio: ["ignore", "pipe", "pipe"],
```

File: test/js/bun/shell/bunshell.test.ts [hunk 3]
```diff
@@ -831,3 +832,61 @@ function sentinelByte(buf: Uint8Array): number {
...
+  );
+  await writeFile(
+    join(package_dir, "index.ts"),
+    `console.log(process.env.NODE_ENV); console.log(process.env.AWESOME);`,
+  );
+  await writeFile(join(package_dir, ".env.production"), `AWESOME=production`);
+  await writeFile(join(package_dir, ".env.development"), `AWESOME=development`);
+
+  const { stdout, stderr, exited } = Bun.spawn({
+    cmd: [bunExe(), "run", "prod"],
```
Comment (Jarred-Sumner) [inline test/js/bun/shell/bunshell.test.ts:850]:
```suggestion
    cmd: [bunExe(),  "--shell=bun", "run", "prod"],
```

File: test/js/bun/shell/bunshell.test.ts [hunk 4]
```diff
@@ -831,3 +832,61 @@ function sentinelByte(buf: Uint8Array): number {
   }
   throw new Error("No sentinel byte");
 }
+
+test("shell as script runner propagates env vars: bun run <foo>", async () => {
```
Comment (Jarred-Sumner) [inline test/js/bun/shell/bunshell.test.ts:836]: this should be condensed into a `test.each` for the `bun run` permutation and the non-bun run permutation

**Comment by Jarred-Sumner** [comment]
I think this bug is actually the shell not loading environment variables, and this fix causes a different bug where it no longer loads .env files in bun run at the right time

**Comment by paperclover** [comment]
superceded by #9642

**Pull request closed without merging**